synchronization/src/main/java/de/cyface/synchronization/MeasurementContentProviderClient.java

```diff
@@ -38,27 +41,74 @@
-     * 
+     *
      * @param measurementIdentifier The device wide unqiue identifier of the measurement to serialize.
      * @param client
      */
-    MeasurementContentProviderClient(final long measurementIdentifier, final @NonNull ContentProviderClient client) {
+    public MeasurementContentProviderClient(final long measurementIdentifier, final @NonNull ContentProviderClient client) {
         this.measurementIdentifier = measurementIdentifier;
         this.client = client;
     }
 
     /**
-     * Loads all the geo locations for the measurement.
+     * Loads a page of the geo locations for the measurement.
      *
+     * @param offset The start index of the first geoLocation to load within the measurement
+     * @param limit  The number of GeoLocations to load, recommended: {@code DATABASE_QUERY_LIMIT}
      * @return A <code>Cursor</code> on the geo locations stored for the measurement.
      * @throws RemoteException If the content provider is not accessible.
      */
-    Cursor loadGeoLocations() throws RemoteException {
-        return client.query(MeasuringPointsContentProvider.GPS_POINTS_URI,
-                new String[] {GpsPointsTable.COLUMN_GPS_TIME, GpsPointsTable.COLUMN_LAT, GpsPointsTable.COLUMN_LON,
-                        GpsPointsTable.COLUMN_SPEED, GpsPointsTable.COLUMN_ACCURACY},
-                GpsPointsTable.COLUMN_MEASUREMENT_FK + "=?",
-                new String[] {Long.valueOf(measurementIdentifier).toString()}, null);
+    public Cursor loadGeoLocations(final int offset, final int limit) throws RemoteException {
+
+        final Uri uri = MeasuringPointsContentProvider.GPS_POINTS_URI;
+        final String[] projection = new String[]{GpsPointsTable.COLUMN_GPS_TIME, GpsPointsTable.COLUMN_LAT, GpsPointsTable.COLUMN_LON,
+                GpsPointsTable.COLUMN_SPEED, GpsPointsTable.COLUMN_ACCURACY};
+        final String selection = GpsPointsTable.COLUMN_MEASUREMENT_FK + "=?";
+        final String[] selectionArgs = new String[]{Long.valueOf(measurementIdentifier).toString()};
+
+        /* For some reason this does not work (tested on N5X) so we always use the workaround implementation
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Bundle queryArgs = new Bundle();
+            queryArgs.putString(android.content.ContentResolver.QUERY_ARG_SQL_SELECTION, selection);
+            queryArgs.putStringArray(android.content.ContentResolver.QUERY_ARG_SQL_SELECTION_ARGS, selectionArgs);
+            queryArgs.putInt(ContentResolver.QUERY_ARG_OFFSET, offset);
+            queryArgs.putInt(ContentResolver.QUERY_ARG_LIMIT, limit);
+            return client.query(uri, projection, queryArgs, null);
+        }*/
+
+        // Backward compatibility workaround
+        return client.query(uri, projection, selection, selectionArgs, GpsPointsTable.COLUMN_MEASUREMENT_FK + " ASC limit " + limit + " offset " + offset);
```

Comment:
Am besten mal mit einem Kommentar und einem Link zum relevaten Thread auf StackOverflow erklären: https://stackoverflow.com/questions/12640480/specifying-limit-offset-for-contentprovider-queries?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa